chat_shell/chat_shell/agents/graph_builder.py

```
@@ -650,6 +872,60 @@ async def stream_tokens(
+
+            # Construct error message for LLM
+            error_message = TOOL_CALL_TRUNCATION_ERROR_TEMPLATE.format(
+                reason=e.reason,
+                attempt=_truncation_retry_count + 1,
+                max_attempts=self.max_truncation_retries,
+            )
+
+            # Add error message to conversation history
+            recovery_messages = list(lc_messages) + [
+                HumanMessage(content=error_message)
+            ]
```

Ruff (0.14.14)

[warning] Consider iterable unpacking instead of concatenation

Replace with iterable unpacking

(RUF005)